### PR TITLE
[chore] do not update PRs if already open with no diff on helm chart

### PR DESCRIPTION
**Pull request by atoulme**
We have a PR [open](https://github.com/signalfx/splunk-otel-collector-chart/pull/1094) for the last few weeks that would be recreated every day based off the cron job updating the PR. Because the workflow creates the PR and then updates it, it forgets the previous state and forces update the PR.

* We get a notification a day of the PR refresh
* We lose work on the PR, such as fixing tests.

To fix this, we want to avoid recreating the PR if no changes are detected after performing the update against the helm-chart folder.

### CODE REVIEW

File: .github/workflows/update_docker_images.yaml [hunk 1]
```diff
@@ -50,9 +52,15 @@ jobs:
 
           make update-docker-image FILE_PATH=${{ matrix.yaml_file_path }} QUERY_STRING='${{ matrix.yaml_value_path }}' DEBUG_MODE=$DEBUG_MODE
 
+      - name: Check if PR is already open
+        id: check_if_pr_open
+        run: |
+          git diff --exit-code master..update-${{ matrix.name }} -- helm-charts
```
Comment (jinja2) [inline .github/workflows/update_docker_images.yaml:58]:
```suggestion
          git diff --exit-code main..update-${{ matrix.name }} -- helm-charts
```

File: .github/workflows/update_docker_images.yaml [hunk 2]
```diff
@@ -50,9 +52,15 @@ jobs:
 
           make update-docker-image FILE_PATH=${{ matrix.yaml_file_path }} QUERY_STRING='${{ matrix.yaml_value_path }}' DEBUG_MODE=$DEBUG_MODE
 
+      - name: Check if PR is already open
```
Comment (jinja2) [inline .github/workflows/update_docker_images.yaml:55]: Can you clarify what is the motivation for doing this? We don't want open PR to be updated on changes to the base branch/main? I don't see any harm in it, so I am probably missing something.

File: .github/workflows/update_docker_images.yaml [hunk 3]
```diff
@@ -50,9 +52,15 @@ jobs:
...
+        id: check_if_pr_open
+        run: |
+          git diff --exit-code main..update-${{ matrix.name }} -- helm-charts
```
Comment (jinja2) [inline .github/workflows/update_docker_images.yaml:58]: [default is to run shell with set -e](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell), so I think on diff or if the branch doesn't exist, the job will simply fail here.

File: .github/workflows/update_docker_images.yaml (hunk 2 above)
Comment (atoulme) [reply .github/workflows/update_docker_images.yaml:55]: Clarified in the description of the PR.

File: .github/workflows/update_docker_images.yaml (hunk 3 above)
Comment (atoulme) [reply .github/workflows/update_docker_images.yaml:58]: makes sense, I'll update. Thanks!

**Review by jvoravong — APPROVED** [review]

**Pull request merged**